qiskit_experiments/library/randomized_benchmarking/rb_experiment.py

```diff
@@ -214,25 +230,68 @@ def _generate_circuit(
                 circuits.append(rb_circ)
         return circuits
 
-    def _get_circuit_metadata(self, circuit):
-        if circuit.metadata["experiment_type"] == self._type:
-            return circuit.metadata
-        if circuit.metadata["experiment_type"] == ParallelExperiment.__name__:
-            for meta in circuit.metadata["composite_metadata"]:
-                if meta["physical_qubits"] == self.physical_qubits:
-                    return meta
-        return None
+    @classmethod
+    def _default_transpile_options(cls) -> Options:
+        options = super()._default_experiment_options()
+
+        # Assumes typical basis gates for superconduncting processor
+        options.basis_gates = ["sx", "rz", "x", "cx", "measure"]
+
+        return options
+
+    def _finalize(self):
+        super()._finalize()
+
+        # Set constraints to compute basis gates EPGs from an estimated EPC
+        gate_error_ratio = self.experiment_options.gate_error_ratio
+
+        if gate_error_ratio == "default":
+            num_qubit_default_basis = {
+                1: {"sx": 1.0, "rz": 0.0, "x": 1.0},
+                2: {"cx": 1.0},
+            }
+            qubits = tuple(self.physical_qubits)
+            gate_error_ratio = {
+                (qubits, k): v for k, v in num_qubit_default_basis.get(len(qubits), {}).items()
+            }
+
+        if gate_error_ratio != "skip":
+
+            gates_in_estimates = set(q_gate_tup[1] for q_gate_tup in gate_error_ratio.keys())
+            # Validate if assumed gates are a part to basis gates in the transpile options
+            basis_gates = set(self.transpile_options.basis_gates)
+            if not basis_gates.issuperset(gates_in_estimates):
+                raise ValueError(
+                    f"Assumed gates {gates_in_estimates} is not valid "
+                    f"subset of basis gates {basis_gates}."
+                )
+
+        # Directly copy the value to experiment data metadata via instance state
+        self._gate_error_ratio = gate_error_ratio
```

Comment:
I think this should be `self.set_experiment_options(gate_error_ratio=gate_error_ratio)` to update the experiment option.

Reply:
Umm... I wanted to avoid implicitly updating experiment option when it is "default".

Reply:
Ah you're right. I'll update the code.

Reply:
Done in https://github.com/Qiskit/qiskit-experiments/pull/762/commits/e53521b33e7d42e166c153e9ab6b2746aefdc569 Thanks!